async_simple/coro/test/LazyTest.cpp

```diff
@@ -1351,15 +1351,13 @@ TEST_F(LazyTest, testBatchedcollectAll) {
 
 TEST_F(LazyTest, testDetach) {
     int count = 0;
-    {
-        executors::SimpleExecutor e1(1);
-        auto test1 = [&count]() -> Lazy<int> {
-            count += 2;
-            co_return count;
-        };
-        test1().via(&e1).detach();
-    }
-
+    executors::SimpleExecutor e1(1);
+    auto test1 = [&count]() -> Lazy<int> {
+        count += 2;
+        co_return count;
+    };
+    test1().via(&e1).detach();
+    std::this_thread::sleep_for(1s);
```

Comment:
It is not good to sleep to wait for it gets done.

The general solution is to wrap this one into something like future we could wait.